utilities/rest-helper-factory.js

```diff
@@ -585,6 +590,285 @@ module.exports = function(logger, mongoose, server) {
+                  message:
+                    'The authentication header was missing/malformed, or the token has expired.'
+                },
+                {
+                  code: 404,
+                  message: 'There was no resource found with that ID.'
+                },
+                { code: 500, message: 'There was an unknown error.' },
+                {
+                  code: 503,
+                  message: 'There was a problem with the database.'
+                }
+              ]
+            },
+            policies: policies
+          },
+          response: {
+            // TODO: add a response schema if needed
+            // failAction: config.enableResponseFail ? 'error' : 'log',
+            // schema: model.readModel ? model.readModel : Joi.object().unknown().optional()
+          }
+        }
+      })
+    },
+
+    /**
+     * Creates an endpoint for DELETE /RESOURCE/
+     * @param server: A Hapi server.
+     * @param model: A mongoose model.
+     * @param options: Options object.
+     * @param logger: A logging object.
+     */
+    // TODO: handle partial recovers (return list of ids that failed/were not found)
+    generateRecoverManyEndpoint: function(server, model, options, logger) {
+      // This line must come first
+      validationHelper.validateModel(model, logger)
+      const Log = logger.bind(chalk.yellow('RecoverMany'))
+
+      const collectionName = model.collectionDisplayName || model.modelName
+      if (config.logRoutes) {
+        Log.note('Generating Recover Many endpoint for ' + collectionName)
+      }
+
+      options = options || {}
+
+      let resourceAliasForRoute
+
+      if (model.routeOptions) {
+        resourceAliasForRoute = model.routeOptions.alias || model.modelName
+      } else {
+        resourceAliasForRoute = model.modelName
+      }
+
+      const handler = HandlerHelper.generateRecoverHandler(model, options, Log)
+
+      let payloadModel = null
+      /* if (config.enableSoftDelete) {
+        payloadModel = Joi.alternatives().try(
+          Joi.array().items(
+            Joi.object({
+              _id: Joi.objectId()
+            })
+          ),
+          Joi.array().items(Joi.objectId())
+        )
+      } else {
+        payloadModel = Joi.array().items(Joi.objectId())
+      } */
+
+      payloadModel = Joi.alternatives().try(
+        Joi.array().items(
+          Joi.object({
+            _id: Joi.objectId()
+          })
+        ),
+        Joi.array().items(Joi.objectId())
+      )
+
+      if (!config.enablePayloadValidation) {
+        payloadModel = Joi.alternatives().try(payloadModel, Joi.any())
+      }
+
+      let auth = false
+      let recoverManyHeadersValidation = Object.assign(headersValidation, {})
+
+      if (config.authStrategy && model.routeOptions.recoverAuth !== false) {
+        auth = {
+          strategy: config.authStrategy
+        }
+
+        const scope = authHelper.generateScopeForEndpoint(model, 'recover', Log)
```

Comment:
This needs to be `const scope = authHelper.generateScopeForEndpoint(model, 'update', Log)` as we are treating this as an update action.